jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -1481,6 +1480,12 @@ private DynamicCapacity(List<RetainableByteBuffer> buffers, ByteBufferPool.Sized
                 throw new IllegalArgumentException("must always retain if cannot aggregate");
         }
 
+        private void checkNotReleased()
+        {
+            if (getRetained() <= 0)
+                throw new IllegalStateException("Already released");
```

Comment:
```suggestion
            if (LOG.isDebugEnabled() && getRetained() <= 0)
                throw new IllegalStateException("Already released");
```

Reply:
If we ever re-introduce this bug by accident, only doing this check when debug logs are enabled means the tests will let it slip.